src/compiler/crystal/macros/methods.cr

```diff
@@ -1654,6 +1656,17 @@ module Crystal
       end
     end
 
+    def self.including_types(type)
+      if type.is_a? NonGenericModuleType
```

Comment:
I think this logic also applies to `GenericModuleType` (but you'll have to do it in a separate branch with duplicate logic).